cli/src/command/generate_config.rs

```diff
@@ -104,30 +102,50 @@ impl GenerateConfig {
     }
 
     /// If no port is provided by user, uses `DEFAULT_EXONUM_LISTEN_PORT`.
-    fn parse_external_address(input: &str) -> Result<SocketAddr, Error> {
-        if let Ok(address) = input.parse() {
-            Ok(address)
-        } else {
-            let ip_address = input.parse()?;
-            Ok(SocketAddr::new(ip_address, DEFAULT_EXONUM_LISTEN_PORT))
+    fn resolve_peer_address(
+        hostname: &str,
+        is_host_with_port: &mut bool,
+    ) -> Result<SocketAddr, Error> {
+        match hostname.to_socket_addrs() {
+            Ok(mut addrs) => addrs.next().ok_or_else(|| {
+                Error::msg(format!(
+                    "No one IP address is related to the domain: {}",
+                    hostname
+                ))
+            }),
+            Err(e) => {
```

Comment:
Nit: IMO, it could make sense to bring the condition here:

```rust
Err(e) if e.kind() == ErrorKind::InvalidInput && !*is_host_with_port => {
    // ...
}
Err(e) => Err(Error::from(e)),
```

AFAIK, the borrow checker no longer emits "by-ref & by-move" error, which would've prevented from doing this with old Rust versions.